### DISCUSSION TITLE
Alias Analysis

### DISCUSSION THREAD

**Discussion by lbettels** [soot-oss/SootUp#817] [Q&A] [answered]
I was wondering if someone has experience of performing Alias Analysis in SootUp.
For Soot, i know multiple tools like IDEal or Boomerang exist, but to the best of my knowledge they only work with Soot, as they dont have SootUp connectors yet, and I dont see any sense in using tools from 8 years ago when it is stated that for new projects SootUp is recommended.
Maybe someone could point an alias analysis tool out to me that is integrated with SootUp, or point me to a research paper that helps with fitting flow functions regarding the alias analysis for the IDE/IFDS tabulation problem.
I am especially interested in interprocedural analysis, otherwise i would just port the intraprocedural analysis form the original Soot, but intraprocedural is not good enough for my use case.
Any help is greatly appreciated.
Cheers!

**Comment by stschott** [comment]
We have some test cases within the sootup.analysis module that showcase how to implement an interprocedural analysis und the sootup.analysis module. Maybe this reference can help you with your task.

**Reply by swissiety** [reply]
as a sidenote: main parts of intraprocedural analysis are ported already - currently in develop

**Reply by lbettels** [reply]
Thank you for the quick answer.
I use the interprocedural analysis classes already, as i ported the [TaintAnalysis framework from Soot](https://github.com/soot-oss/TaintAnalysis) to SootUp.
My problem is that i currently have no way of using Boomerang to find the Aliases, as i am using SootUp and not the old Soot.
So my question rather should have been: Are there already available implementations of an algorithm that allows me to get information about the aliases of a Value at a Statement for SootUp?
Is it maybe possible to translate a new SootMethod to a old SootMethod and back?
Then i could just convert the new SootMethod to the old SootMethod and plug it into Boomerang.

**Reply by JonasKlauke** [reply]
The Points-to-Analysis is still in development in the Qilin PR.
SootUp is still quite new and many features or tools working on Soot are still in the construction.
It needs some time for people to shift their focus to the new SootUp which could result in more functionalities or tools for SootUP.
I know that someone is already working on porting Boomerang to SootUp.
We are happy for everyone that helps to further improve/develop SootUp or who develops new tools building on SootUp

**Reply by lbettels** [reply, marked as answer]
Thanks for your reply. I think the Qilin PR already has an Alias Analysis so i will check that out. And if Boomerang for SootUp would be released in the coming months that would be even more fantastic.
Cheers!
